types/bitbucket-definition.json

```diff
@@ -10,20 +10,42 @@
   },
   "namespace_definition": {
     "requirement": "required",
-    "note": "The namespace is the user or organization. It is not case sensitive and must be lowercased.",
-    "native_name": "user or organization",
-    "is_case_sensitve": false
+    "note": "The namespace is the user or organization (for Bitbucket Cloud), or the Project Key (for Bitbucket Server/Data Center). It is not case sensitive and must be lowercased.",
+    "native_name": "user, organization, or project key",
+    "case_sensitive": false,
+    "normalization_rules": [
+      "It is not case sensitive and must be lowercased."
+    ]
   },
   "name_definition": {
+    "requirement": "required",
     "note": "The name is the repository name. It is not case sensitive and must be lowercased.",
     "native_name": "repository name",
-    "is_case_sensitve": false
+    "case_sensitive": false,
+    "normalization_rules": [
+      "It is not case sensitive and must be lowercased."
+    ]
   },
   "version_definition": {
-    "note": "The version is a commit or tag.",
+    "requirement": "required",
+    "note": "The version is an immutable commit hash (full or short) or a tag. Branch names (e.g., 'main') should not be used as they are mutable pointers.",
```

Comment:
This doesn't work. Tags are also mutable. Software that works with PURLs cannot reasonably knowing whether a reference is intended to be mutable.